pom.xml

```diff
@@ -582,19 +582,20 @@
               <version>${checkstyle.version}</version>
             </dependency>
           </dependencies>
-          <configuration>
-            <configLocation>checkstyle.xml</configLocation>
-            <suppressionsLocation>checkstyleSuppressions.xml</suppressionsLocation>
-            <encoding>UTF-8</encoding>
-            <consoleOutput>true</consoleOutput>
-            <failOnViolation>true</failOnViolation>
-            <includeResources>false</includeResources>
-            <includeTestSourceDirectory>true</includeTestSourceDirectory>
-          </configuration>
           <executions>
             <execution>
               <id>checkstyle</id>
               <phase>validate</phase>
+              <configuration>
+                <configLocation>checkstyle.xml</configLocation>
```

Comment:
this way you are disabling the minimal checkstyle configuration I added to prevent the presence of '@author' tags.
As your new configuration is applied only to some limited list of packages you are implicitly disabling that check on a part of the codebase.

Please  add a new   '<execution>' of the plugin with your new checkstyle.xml and checkstyleSuppressions.xml

Please test manually that if you add a '@author' javadoc tag anywhere in code the PR validation fails

Reply:
@eolivelli 
- Done.use the separate `checkstyle-strict.xml` for the `zookeeper-server` module.
- I check that our code base doesn't have the **'@author'** tag

Reply:
It is reasonable to use a `checkstyle-strict.xml` covering the simple config in pkgs we want to enable more rules.

For "I check that our code base doesn't have the '@author' tag", I think you should also test if a '@author' tag exists, the build fails with correct reason.

Reply:
- @TisonKun  Thanks for the notice.
- I check the rule for the "the '@author' tag" 

> <module name="TodoComment">
            <!-- Checks that disallowed strings are not used in comments.  -->
            <property name="format" value="(@author)" />
        </module>

and found it doesn't work.search around and don't find a valid way to enforce no `author` tag in the javaDoc.Cc @eolivelli

Reply:
Hi @maoling , you can try out enable the former simple checkstyle configuration in parent pom.xml and enable strict checkstyle configuration in, for example, zookeeper-server pkg. Make sure the strict checkstyle configuration contains rules in the simple one including forbid '@author' tag. I would provide a patch you can make use of in this week if you stuck.

Reply:
@TisonKun 
Nothing can make me get stuck, except the beautiful girls;)

Reply:
@maoling you can take a look at ZOOKEEPER-3465 and #1022 #1023 where I show you how to enable checkstyle configuration per pkg and why I stand "no need" for extra suppression configuration. 

You can rebase the subtasks enable it on zookeeper-server(i.e., this thread) on #1022 . Feel free to share you opinion because I did change the checkstyle configuration choice.